opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -226,79 +258,69 @@ message HistogramValue {
-    // [bucket_bounds[i-1], bucket_bounds[i]) for 0 < i < N-1
-    // [bucket_bounds[i], +infinity) for i == N-1
-    message Explicit {
-      // The values must be strictly increasing and > 0.
-      repeated double bounds = 1;
-    }
-
-    // TODO: If OpenMetrics decides to support (a, b] intervals we should add
-    // support for these by defining a boolean value here which decides what
-    // type of intervals to use.
-  }
-
-  // Don't change bucket boundaries within a TimeSeries if your backend doesn't
-  // support this.
-  BucketOptions bucket_options = 5;
-
+  // Bucket contains values for a bucket.
   message Bucket {
-    // The number of values in each bucket of the histogram, as described in
-    // bucket_bounds.
+    // The number of values in each bucket of the histogram, as described by
+    // bucket_options.
     int64 count = 1;
 
     // Exemplars are example points that may be used to annotate aggregated
     // Histogram values. They are metadata that gives information about a
     // particular value added to a Histogram bucket.
     message Exemplar {
-      // Value of the exemplar point. It determines which bucket the exemplar
-      // belongs to.
+      // Value of the exemplar point. It determines which bucket the exemplar belongs to.
+      // If bucket_options define bounds for this bucket then this value must be within
+      // the defined bounds.
       double value = 1;
 
       // timestamp_unixnano is the moment when this exemplar was recorded.
       // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
       sfixed64 timestamp_unixnano = 2;
 
-      // Contextual information about the example value.
-      map<string, string> attachments = 3;
+      // exemplar_attachments are contextual information about the example value.
+      // Keys in this list must be unique.
+      repeated StringKeyValuePair attachments = 3;
     }
 
-    // Exemplars are example points that may be used to annotate aggregated
-    // Histogram values.
+    // exemplar is an optional representative value of the bucket.
     Exemplar exemplar = 2;
   }
 
-  // The sum of the values in the Bucket counts must equal the value in the
-  // count field of the histogram.
-  repeated Bucket buckets = 6;
+  // buckets is an optional field contains the values of histogram for each bucket.
+  //
+  // The sum of the values in the buckets "count" field must equal the value in the
+  // count field of HistogramValue.
+  //
+  // The number of elements in buckets array must be by one greater than the
+  // number of elements in bucket_bounds array.
+  //
+  // Note: if HistogramTimeSeries.bucket_options defines bucket bounds then this field
+  // must also be present and number of elements in this field must be equal to the
+  // number of buckets defined by bucket_options.
+  repeated Bucket buckets = 5;
+}
+
+// StringKeyValuePair is a pair of key/value strings.
+message StringKeyValuePair {
```

Comment:
Add a todo to look to unify this with label/resource etc.

Reply:
Done.